lib/upgrade_html.js

```diff
@@ -43,9 +44,41 @@ function upgradeHtml(filename) {
     upgradeDataBoundTemplate($, autoBindTemplate);
   });
 
+  upgradeGlobalCss($);
+
   $('polymer-element').each(function(_ignored, polyElem) {
     var elemName = polyElem.attribs.name;
 
+    var domModule = $('<dom-module>');
+    domModule.attr('id', elemName);
+    domModule.text('\n');
+
+    var templateChildren = $(polyElem).find('template');
+    var template;
+    if (templateChildren[0]) {
+      template = templateChildren[0];
+
+      // Migrate styles up to be a direct child of dom-module
+      $(template.children[0]).find('style').each(function(_ignored, styleElem) {
+        domModule.append('  ');
+        var before = styleElem.previousSibling;
+        if (before.type === 'text') {
+          $(before).remove();
+        }
+        domModule.append($(styleElem));
+        domModule.append('\n');
+      });
+
+      // Then move all templates in after
+      templateChildren.each(function(_ignored, templateChild) {
+        domModule.append('  ');
+        domModule.append(templateChild);
+        domModule.append('\n');
+      });
+      upgradeElementCss($, polyElem, domModule, template);
```

Comment:
The addition of this line is the only change in this block, the rest is just moving code around so that the operations are done in the right order.